packages/twenty-server/src/engine/metadata-modules/field-metadata/utils/resolve-overridable-string.util.ts

```diff
@@ -17,12 +18,28 @@ export const resolveOverridableString = (
     return fieldMetadata[labelKey] ?? '';
   }
 
-  const translationValue =
-    // @ts-expect-error legacy noImplicitAny
-    fieldMetadata.standardOverrides?.translations?.[locale]?.[labelKey];
+  if (labelKey === 'icon' && isDefined(fieldMetadata.standardOverrides?.icon)) {
+    return fieldMetadata.standardOverrides.icon;
+  }
+
+  if (
+    isDefined(fieldMetadata.standardOverrides?.translations) &&
+    isDefined(locale) &&
+    labelKey !== 'icon'
+  ) {
+    const translationValue =
+      fieldMetadata.standardOverrides.translations[locale]?.[labelKey];
+
+    if (isDefined(translationValue)) {
+      return translationValue;
+    }
+  }
 
-  if (isDefined(translationValue)) {
-    return translationValue;
+  if (
```

Comment:
then we care for this edge case:
- for the SOURCE_LOCALE (which is always 'en' for now), we do not look for translations but look for the value at root of standardOverrides.

Why? We could have stored it in 'en' locale. However, this would be inconsitent with custom fields behavior where we rename the field for all languages when we rename it. This edge case will be resolved with a better translation UI later in FE